File: examples_android/app/src/main/java/com/confirmit/testsdkapp/layouts/simpleCardLayout/items/SimpleCardTextItem.kt
package com.confirmit.testsdkapp.layouts.simplecardlayout.items

import android.view.View
import android.widget.EditText
import com.confirmit.mobilesdk.ui.questions.TextQuestion
import com.confirmit.testsdkapp.R

class SimpleCardTextItem(view: View) : SimpleCardItem(view) {

    private lateinit var question: TextQuestion
    private val txtContent: EditText = view.findViewById(R.id.editText)

    fun setupQuestion(question: TextQuestion) {
        this.question = question
        txtHeader.text = question.text.get()
        txtContent.setText(question.getValue())

        handleValidation(question.errors)
    }

    override fun onShow(): Boolean {
        txtContent.requestFocus()
        return false
    }

    override fun onNext(): Boolean {
        question.setValue(txtContent.text.toString())
        return validate(question)
    }

    override fun onBack(): Boolean {
        question.setValue(txtContent.text.toString())
        return true
    }
}

File: examples_ios/MyApp/Sources/SurveyLayouts/SimpleCardLayout/Items/SimpleCardTextItem.swift
import Foundation
import ConfirmitMobileSDK
import UIKit

class SimpleCardTextItem: SimpleCardItem {
    @IBOutlet var contentContainer: UIView!
    @IBOutlet var txtContent: UITextView!
    
    var question: TextQuestion!
    
    override func awakeFromNib() {
        super.awakeFromNib()
        
        contentContainer.layer.cornerRadius = 12
        contentContainer.layer.borderColor = SimpleCardColor.border.cgColor
        contentContainer.layer.borderWidth = 1
        txtContent.tintColor = SimpleCardColor.button
        txtContent.font = UIFont.systemFont(ofSize: 16)
    }
    
    func setupQuestion(question: TextQuestion) {
        self.question = question
        lblHeader.text = question.text.get()
        txtContent.text = question.getValue()
        
        handleValidation(errors: question.errors)
    }
    
    override func onShow() -> Bool {
        txtContent.becomeFirstResponder()
        return false
    }
    
    override func onNext() -> Bool {
        question.setValue(value: txtContent.text)
        return validate(question: question)
    }
    
    override func onBack() -> Bool {
        question.setValue(value: txtContent.text)
        return true
    }
}
